python/activator/middleware_interface.py

```diff
@@ -282,20 +278,55 @@ def _prep_collections(self):
         #             'refcats'],
         #     output=self.output_collection)
         # The below is taken from SimplePipelineExecutor.prep_butler.
-        output_run = f"{self.output_collection}/{self.instrument.makeCollectionTimestamp()}"
+        self.output_run = f"{self.output_collection}/{self.instrument.makeCollectionTimestamp()}"
         self.butler.registry.registerCollection(self.instrument.makeDefaultRawIngestRunName(),
                                                 CollectionType.RUN)
-        self.butler.registry.registerCollection(output_run, CollectionType.RUN)
+        self.butler.registry.registerCollection(self.output_run, CollectionType.RUN)
         self.butler.registry.registerCollection(self.output_collection, CollectionType.CHAINED)
-        collections = [self.calibration_collection,
+        collections = [self.instrument.makeCollectionName("defaults"),
+                       self.instrument.makeCalibrationCollectionName(),
                        self.instrument.makeDefaultRawIngestRunName(),
                        'refcats',
-                       output_run]
+                       self.output_run]
         self.butler.registry.setCollectionChain(self.output_collection, collections)
 
-        # Refresh butler after configuring it, to ensure all required
-        # dimensions are available.
-        self.butler.registry.refresh()
+        # Need to create a new butler with all the output collections.
+        self.butler = Butler(butler=self.butler,
+                             collections=[self.output_collection],
+                             # TODO DM-34202: hack around a middleware bug.
+                             run=None)
+
+    def _prep_pipeline(self, visit: Visit) -> None:
+        """Setup the pipeline to be run, based on the configured instrument and
+        details of the incoming visit.
+
+        Parameters
+        ----------
+        visit : Visit
+            Group of snaps from one detector to prepare the pipeline for.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if there is no AP pipeline file for this configuration.
+            TODO: could be a good case for a custom exception here.
+        """
+        # TODO: we probably want to be able to configure this per-instrument?
+        # TODO: DM-33453 will remap ap_pipe/pipelines to use getName convention
+        camera_paths = {"DECam": "DarkEnergyCamera", "HSC": "HyperSuprimeCam"}
+        try:
+            camera_path = camera_paths[self.instrument.getName()]
+        except KeyError:
+            raise RuntimeError(f"No ApPipe.yaml defined for camera {self.instrument.getName()}")
+        # TODO: DECam test data uses `panstarrs/gaia` for the refcat names.
+        yaml_file = "ApPipe.yaml" if self.instrument.getName() != "DECam" else "ApPipe_hits2015-3.yaml"
```

Comment:
[Umm... no](https://github.com/lsst/ap_pipe/blob/main/pipelines/DarkEnergyCamera/ApPipe.yaml). `ApPipe_hits2015-3.yaml` is of historical interest and probably won't work with the current Stack.

If the goal is to work around the quirks of `ap_verify` datasets, please configure the pipeline directly. Aside from the refcats, this should just be a matter of setting [the parameters](https://github.com/lsst/ap_pipe/blob/main/pipelines/ApPipe.yaml#L14-L28). Though I wonder if it would be more expedient to just copy the datasets' `ApPipe.yaml` files for now.